libsolidity/ast/AST.h

```diff
@@ -1241,6 +1245,12 @@ class EventDefinition: public CallableDeclaration, public StructurallyDocumented
 	{
 	}
 
+	struct CompareBySignature
+	{
+		std::string const eventSignature(EventDefinition const* _event) const;
```

Comment:
We don't typically use `const` in cases like this. It's a copy anyway so no reason to make the caller make another one if it wants to modify it.

```suggestion
		std::string eventSignature(EventDefinition const* _event) const;
```